cmd/kubeadm/app/util/config/common.go

```diff
@@ -70,11 +70,17 @@ func DetectUnsupportedVersion(b []byte) error {
 	oldKnownAPIVersions := map[string]string{
 		"kubeadm.k8s.io/v1alpha1": "v1.11",
 	}
+	knownKinds := map[string]bool{}
 	// If we find an old API version in this gvk list, error out and tell the user why this doesn't work
 	for _, gvk := range gvks {
 		if useKubeadmVersion := oldKnownAPIVersions[gvk.GroupVersion().String()]; len(useKubeadmVersion) != 0 {
 			return fmt.Errorf("your configuration file uses an old API spec: %q. Please use kubeadm %s instead and run 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.", gvk.GroupVersion().String(), useKubeadmVersion)
 		}
+		knownKinds[gvk.Kind] = true
+	}
+	// MasterConfiguration and NodeConfiguration are mutually exclusive, error if both are specified
```

Comment:
I'm torn between implementing this check, that is formally correct, a being a little bit permissive and simply ignore everything is not in scope of the current operation

Reply:
Having the complete recipe for creating a K8s cluster in a single YAML file can be good for someone. Yet, this can encourage sloppiness on part of some admins and tools. Furthermore, the code may be more streamlined here and there if we avoid having init and join configs at the same time.

Reply:
Yeah, if both are given we have no idea on what kubeadm should actually do, these should be mutually exclusive, otherwise we'll find weird bugs.